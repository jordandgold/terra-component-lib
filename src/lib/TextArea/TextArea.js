import React, { Component } from "react";
import "./TextArea.scss";
import PropTypes from "prop-types";

class TextArea extends Component {
  render() {
    return (
      <div className="ter-form-item">
        <label className="ter-form-item__label">{this.props.label}</label>
        <textarea
          defaultValue={this.props.value}
          onChange={e => this.props.inputChange(e)}
          name={this.props.name}
          className="ter-textarea"
        />
      </div>
    );
  }
}

export default TextArea;

TextArea.propTypes = {
  label: PropTypes.string.isRequired,
  inputChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};
