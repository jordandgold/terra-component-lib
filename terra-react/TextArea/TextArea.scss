@import "../global-styles/terra.scss";

.ter-textarea {
  @include input-base();
  min-height: 120px;
}
