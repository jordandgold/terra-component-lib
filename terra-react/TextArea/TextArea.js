import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import React, { Component } from "react";
import "./TextArea.scss";

var TextArea =
/*#__PURE__*/
function (_Component) {
  _inherits(TextArea, _Component);

  function TextArea() {
    _classCallCheck(this, TextArea);

    return _possibleConstructorReturn(this, _getPrototypeOf(TextArea).apply(this, arguments));
  }

  _createClass(TextArea, [{
    key: "render",
    value: function render() {
      var _this = this;

      return React.createElement("div", {
        className: "ter-form-item"
      }, React.createElement("label", {
        className: "ter-form-item__label"
      }, this.props.label), React.createElement("textarea", {
        defaultValue: this.props.value,
        onChange: function onChange(e) {
          return _this.props.inputChange(e, _this.props.name);
        },
        name: this.props.name,
        className: "ter-textarea"
      }));
    }
  }]);

  return TextArea;
}(Component);

export default TextArea;