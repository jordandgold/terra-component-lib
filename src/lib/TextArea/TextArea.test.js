import React from "react";
import { shallow } from "enzyme";
import TextArea from "./TextArea";

describe("TextArea", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();
    wrapper = shallow(
      <TextArea
        handleChange={mockHandleChange}
        value="mock value"
        label="mock label"
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange on change", () => {
    const mockObject = {
      target: {
        value: "Hello"
      }
    };
    wrapper.find("textarea").simulate("change", mockObject);

    expect(mockHandleChange).toHaveBeenCalledWith(mockObject);
  });
});
