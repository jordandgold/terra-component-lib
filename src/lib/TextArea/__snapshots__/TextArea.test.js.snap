// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should match the snapshot 1`] = `
<div
  className="ter-input__wrap"
>
  <label
    classame="ter-input__label"
  >
    mock label
  </label>
  <textarea
    defaultValue="mock value"
    onChange={[Function]}
  />
</div>
`;
